satellite/gc/queue.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package gc
+
+import (
+	"context"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	monkit "gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/auth/signing"
+	"storj.io/storj/pkg/bloomfilter"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/pkg/transport"
+	"storj.io/storj/uplink/piecestore"
+)
+
+var (
+	mon = monkit.Package()
+
+	// Error defines the garbage queue errors class
+	Error = errs.Class("garbage queue error")
+)
+
+// RetainInfo contains info needed for a storage node to retain important data and delete garbage data
+type RetainInfo struct {
+	Filter       *bloomfilter.Filter
+	CreationDate time.Time
+}
+
+// Queue defines the functions of the satellite's garbage collection "queue"
+// TODO: does this need to be an interface?
+type Queue interface {
```

Comment:
Do you even need a queue at this point?

Reply:
do you mean the interface or the data type of queue? if the latter, yeah I don't think so, I think it needs a different name.

Reply:
I mean the interface in general.

Reply:
I don't think so, I'll remove it.